core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java

```diff
@@ -89,6 +90,15 @@
    *
    */
   private boolean isTransactionalTable = true;
+  /**
+   * The format of the fact table.
+   * By default it is carbondata, and we also support other format like CSV
+   */
+  private String format = "carbondata";
```

Comment:
This class should be backward compatible, please make sure you will process it if it is null
And please add @Since("1.4.1") annotation